ion/reader.go

```diff
@@ -159,9 +155,9 @@ type Reader interface {
 	// IsInStruct indicates if the reader is currently positioned in a struct.
 	IsInStruct() bool
 
-	// FieldNameSymbol returns the field name associated with the current value as a SymbolToken.
-	// It returns an error if the current value has no field name.
-	FieldNameSymbol() (*SymbolToken, error)
+	// FieldName returns the field name associated with the current value as a SymbolToken. It returns
+	// nil if there is no current value or the current value has no field name.
```

Comment:
This needs a little more information. When does `err` get populated with something other than `nil`?

Reply:
Updated to:

```
	// FieldName returns the field name associated with the current value as a SymbolToken. It returns
	// nil if there is no current value or the current value has no field name. It returns an error if
	// the `SymbolToken` SID is not found in the symbol table.
```